src/v/datalake/translation/state_machine.h

```diff
@@ -30,6 +30,7 @@ class translation_stm final : public raft::persisted_stm<> {
 
     ss::future<> do_apply(const model::record_batch&) override;
     model::offset max_collectible_offset() override;
+    std::optional<kafka::offset> lowest_pinned_offset() const override;
```

Comment:
maybe it would be worth renaming the two methods to reflect the purpose of them better ?

Reply:
Per our slack conversation, went with:
- max_removable_local_log_offset
- lowest_pinned_data_offset